WordPress/gutenberg
# [RNMobile] Upgrades Gradle to 7.4.2 & Android Gradle Plugin to 7.2.1

## What?

This PR upgrades `react-native-bridge`, `react-native-aztec` & `react-native-editor` Android project's Gradle version to `7.4.2` & Android Gradle Plugin version to `7.2.1`.

Since Gradle update is just a patch, it only contains bug fixes. Here are the full release notes: https://docs.gradle.org/7.4.2/release-notes.html

---

The only major Android Gradle Plugin release since `7.1.1` is `7.2` which includes the following breaking change:

> Starting with AGP 7.2, you can no longer share the same source directory across multiple source sets.
> For example, you can’t use the same test sources for both unit tests and instrumentation tests. To learn more, see Change the default source sets configurations.

As far as I can tell if this is an issue for us it should result in a build time error, so as long as `gutenberg-mobile` [CI](https://github.com/wordpress-mobile/gutenberg-mobile/pull/5002) passes I don't think we need to worry about it.

## Why?


[Gradle composite builds](https://docs.gradle.org/current/userguide/composite_builds.html) require Android Gradle Plugin version to be the same for included builds. Since we are updating WordPress-Android in https://github.com/wordpress-mobile/WordPress-Android/pull/16834 we need to update these projects as well.

## Testing Instructions





* Smoke test the demo Android app
* Make sure the `gutenberg-mobile` PR [CI](https://github.com/wordpress-mobile/gutenberg-mobile/pull/5002) passes

cc @ParaskP7
